### PR TITLE
Un-generic methods related to filteredItem and selectedItem

### PR DESCRIPTION
**Pull request by zj-cs2103**
Un-generic updateFilteredItemList, selectedItemProperty, getSelectedItem and setSelectedItem methods.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/RestockIngredientCommand.java [hunk 1]
```diff
@@ -3,7 +3,8 @@
 import static java.util.Objects.requireNonNull;
 import static seedu.address.logic.parser.CliSyntax.PREFIX_INDEX;
 import static seedu.address.logic.parser.CliSyntax.PREFIX_INGREDIENT_QUANTITY;
-import static seedu.address.model.Model.PREDICATE_SHOW_ALL_ITEMS;
+import static seedu.address.model.IngredientModel.PREDICATE_SHOW_ALL_INGREDIENTS;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_INGREDIENTS;
```
Comment (lingyanhao) [inline src/main/java/seedu/address/logic/commands/RestockIngredientCommand.java:6]: why two separate imports for the same thing??

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 2]
```diff
@@ -5,7 +5,8 @@
 import static seedu.address.logic.parser.CliSyntax.PREFIX_LOYALTY_POINTS;
 import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
 import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
-import static seedu.address.model.Model.PREDICATE_SHOW_ALL_ITEMS;
+import static seedu.address.model.MemberModel.PREDICATE_SHOW_ALL_MEMBERS;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_MEMBERS;
```
Comment (lingyanhao) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:8]: why two separate imports

File: src/main/java/seedu/address/logic/commands/RedoCommand.java [hunk 3]
```diff
@@ -27,7 +26,7 @@ public CommandResult execute(Model model, CommandHistory history) throws Command
         }
 
         model.redoRestaurantBook();
-        model.updateFilteredItemList(PREDICATE_SHOW_ALL_ITEMS, Member.class);
+        model.updateFilteredMemberList(PREDICATE_SHOW_ALL_MEMBERS);
```
Comment (lingyanhao) [inline src/main/java/seedu/address/logic/commands/RedoCommand.java:29]: (Out of topic) - Might need to update all the filtered lists once we implement our own filter operations.

File: src/main/java/seedu/address/model/ModelManager.java [hunk 4]
```diff
@@ -283,60 +287,75 @@ public void commitRestaurantBook() {
...
+    public Staff getSelectedStaff() {
+        return selectedStaff.getValue();
+    }
+
+    @Override
+    public void setSelectedMember(Member member) {
+        if (member != null && !filteredMembers.contains(member)) {
+            throw new ItemNotFoundException();
         }
+        selectedMember.setValue((Member) member);
```
Comment (lingyanhao) [inline src/main/java/seedu/address/model/ModelManager.java]: Why is the explicit cast needed here?

File: src/main/java/seedu/address/model/ModelManager.java [hunk 5]
```diff
@@ -283,60 +287,75 @@ public void commitRestaurantBook() {
...
-            return (T) selectedIngredient.getValue();
-        } else if (clazz == Staff.class) {
-            return (T) selectedStaff.getValue();
-        } else {
-            throw new IllegalArgumentException("Item type not recognised.");
+    public void setSelectedBooking(Booking booking) {
+        if (booking != null && !filteredBookings.contains(booking)) {
+            throw new ItemNotFoundException();
         }
+        selectedBooking.setValue((Booking) booking);
```
Comment (lingyanhao) [inline src/main/java/seedu/address/model/ModelManager.java]: Same as above- why the explicit cast

File: src/main/java/seedu/address/logic/commands/RestockIngredientCommand.java (hunk 1 above)
Comment (zj-cs2103) [reply src/main/java/seedu/address/logic/commands/RestockIngredientCommand.java:6]: fixed

File: src/main/java/seedu/address/logic/commands/EditCommand.java (hunk 2 above)
Comment (zj-cs2103) [reply src/main/java/seedu/address/logic/commands/EditCommand.java:8]: fixed

File: src/main/java/seedu/address/logic/commands/RedoCommand.java (hunk 3 above)
Comment (zj-cs2103) [reply src/main/java/seedu/address/logic/commands/RedoCommand.java:29]: Yes

File: src/main/java/seedu/address/model/ModelManager.java (hunk 5 above)
Comment (zj-cs2103) [reply src/main/java/seedu/address/model/ModelManager.java]: fixed

File: src/main/java/seedu/address/model/ModelManager.java (hunk 4 above)
Comment (zj-cs2103) [reply src/main/java/seedu/address/model/ModelManager.java]: fixed

**Pull request merged**